train.py

```diff
@@ -1,17 +1,67 @@
-from absl import app, flags
+import os
+from datetime import datetime
+
+import wandb
+from absl import app, flags, logging
 from ml_collections.config_flags import config_flags
 from tensorflow import keras
 
 from sagan.dataloader import DataLoader
 from sagan.models.sagan import SelfAttentionGAN
+from sagan.utils import init_wandb, initialize_device
 
 FLAGS = flags.FLAGS
 config_flags.DEFINE_config_file("experiment_configs")
 
 
 def main(_):
+
+    # Get the current timestamp
+    timestamp = datetime.utcnow().strftime("%y%m%d-%H%M%S")
+
+    # Iinitialising Weights and Biases, if wandb is used
+    if FLAGS.experiment_configs.use_wandb:
+        wandb.login()
+        init_wandb(
+            FLAGS.experiment_configs.project_name,
+            f"{FLAGS.experiment_configs.project_name}_{timestamp}",
+            FLAGS.experiment_configs.to_dict(),
+        )
+
+    # Detect strategy
+    strategy = initialize_device()
+    batch_size = FLAGS.experiment_configs.batch_size * strategy.num_replicas_in_sync
+    FLAGS.experiment_configs["batch_size"] = batch_size
+    logging.info(f"Preparing data loader with a batch size of {batch_size}.")
+
     ## Load the data
-    dataset = DataLoader.get_dataset()
+    pipeline_train, pipeline_test = DataLoader.get_dataset(
+        batch_size=FLAGS.experiment_configs.batch_size
+    )
+    logging.info(f"Number of training examples: {pipeline_train.cardinality()}")
+    logging.info(f"Number of validation examples: {pipeline_test.cardinality()}")
+
+    # Setting up callback
+    logging.info("Initializing callbacks.")
+
+    # Model Checkpoint Callback
+    checkpoint_callback = keras.callbacks.ModelCheckpoint(
+        filepath=f"{FLAGS.experiment_configs.checkpoint_filepath}_{timestamp}",
+        save_best_only=True,
```

Comment:
This is probably not required for a GAN model. Instead, the best course of action would be to create a custom callback to version each checkpoint as an [artifact](https://docs.wandb.ai/guides/artifacts). Unfortunately `wandb.keras.WandbCallback()` doesn't automatically save checkpoints yet, but it's gonna be resolved soon. Till then I think that we could create a custom callback inheriting from `keras.callbacks.ModelCheckpoint`.